microsoftgraph/microsoft-graph-explorer-v4
# Track time it takes to get a response from DevX API

## Overview

Add a custom event which allows us to capture the time it takes for a call to our DevX API to return a response.

Application Insights JS SDK offers two methods to use to track events:

- `appInsights.startTrackEvent(eventName)`, to be called just before the tracked call
- `appInsights.stopTrackEvent(eventName, properties)` to be called just after the tracked call

Application Insights will add a custom property `duration` with the API call duration in milliseconds.

## Testing Instructions

* Open Developer Tools on browser then go Network tab
* Fetching samples, code snippets, permissions or autocomplete options should trigger a tracked API call
* Inspect data sent to Application Insights. Below is an example of how the data should look like:
![image](https://user-images.githubusercontent.com/25274795/109765390-d67b0800-7c05-11eb-823f-e5afacbaa8ae.png)
